### PR TITLE
Reorganize build system docs

### PR DESCRIPTION
**Pull request by knelli2**
## Proposed changes

And also move them under `Installation` instead of the `Developers Guide`.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### Further comments

Depends on #6164

### CODE REVIEW

File: docs/Installation/Installation.md [hunk 1]
```diff
@@ -7,7 +7,8 @@ See LICENSE.txt for details.
 \tableofcontents
 
 This page details the installation procedure for SpECTRE on personal computers
-using x86-64 processors. For instructions on installing SpECTRE on clusters
+using x86-64 processors. For configuring SpECTRE, please refer to the \subpage
+spectre_build_system pag. For instructions on installing SpECTRE on clusters
```
Comment (kidder) [inline docs/Installation/Installation.md]: page

**Review by kidder — CHANGES_REQUESTED** [review]

**Comment by knelli2** [comment]
Marking priority because this should be merged before the workshop

**Comment by knelli2** [comment]
@kidder could you take a look at this again?

File: docs/Installation/BuildSystem.md [hunk 2]
```diff
@@ -2,163 +2,57 @@
...
+to specify flags yourself.
 
 ## Commonly Used CMake flags {#common_cmake_flags}
-The following are the most common flags used to control building with
-`CMake`. They are used by
-```
-cmake -D FLAG1=OPT1 ... -D FLAGN=OPTN <SPECTRE_ROOT>
-```
+The following are common flags used to control building SpECTRE with CMake (in
+alphabetical order):
```
Comment (kidder) [inline docs/Installation/BuildSystem.md:55]: while you're touching the file, alphabetize the list (BUILD_DOCS, DOCS_ONLY, the CMAKE_*, SPECTRE_DEBUG)

File: docs/Installation/BuildSystem.md [hunk 3]
```diff
@@ -405,6 +306,148 @@ available to build with `make` or `ninja`:
...
+
+## Adding Source Files {#adding_source_files}
+
+SpECTRE organizes source files into subdirectories of `src` that are
+compiled into libraries.  To add a new source file `FILE.cpp` to an
+existing library in `src/PATH/DIR`, just edit
+`src/PATH/DIR/CMakeLists.txt` and add `FILE.cpp` to the list of files
+in
+```
+set(LIBRARY_SOURCES
```
Comment (kidder) [inline docs/Installation/BuildSystem.md]: this is not true anymore.  we have `spectre_target_sources` and `spectre_target_headers`

File: docs/Installation/BuildSystem.md [hunk 4]
```diff
@@ -405,6 +306,148 @@ available to build with `make` or `ninja`:
...
+- Add the line:
+```
+set(LIBRARY LIB)
+```
+where convention is that `LIB` = `DIR`.  As library names must be
+unique, this is not always possible, in which case the convention is to
+prepend the parent directory to `DIR`.
+- Add the lines
+```
+set(LIBRARY_SOURCES
```
Comment (kidder) [inline docs/Installation/BuildSystem.md]: see above

**Review by kidder — CHANGES_REQUESTED** [review]

**Comment by knelli2** [comment]
@kidder posted a fixup since I edited a couple more things

**Review by kidder — COMMENTED** [review]
looks good, squash

**Review by kidder — APPROVED** [review]

**Pull request merged**